src/sourcemap/sourcemap.zig

```diff
@@ -1522,15 +1522,9 @@ pub fn appendSourceMapChunk(
     start_state.original_line += original_line.value;
     start_state.original_column += original_column.value;
 
-    j.push(
-        appendMappingToBuffer(
-            MutableString.initEmpty(allocator),
-            j.lastByte(),
-            prev_end_state,
-            start_state,
-        ).list.items,
-        allocator,
-    );
+    var str = MutableString.initEmpty(allocator);
+    appendMappingToBuffer(&str, j.lastByte(), prev_end_state, start_state);
+    j.push(str.slice(), allocator);
```

Comment:
`str.slice()` might have a different length than the underlying memory, so when we free it the length won't be correct.